test/sim/abilities/pressure.js

```diff
@@ -85,6 +85,50 @@ describe('Pressure', function () {
 		assert.equal(battle.p2.active[0].getMoveData(Dex.getMove('snatch')).pp, 12);
 		assert.equal(battle.p2.active[1].getMoveData(Dex.getMove('imprison')).pp, 12);
 	});
+
+	it.skip(`should deduct additional PP from Max Moves`, function () {
+		battle = common.createBattle([[
+			{species: 'wynaut', moves: ['darkpulse']},
+		], [
+			{species: 'absol', ability: 'pressure', moves: ['sleeptalk']},
+		]]);
+		battle.makeChoices('move darkpulse dynamax', 'auto');
+		assert.equal(battle.p1.active[0].getMoveData(Dex.getMove('darkpulse')).pp, 22);
+	});
+
+	it.skip(`should deduct additional PP from Z-Moves`, function () {
+		battle = common.gen(7).createBattle([[
+			{species: 'wynaut', item: 'darkiniumz', moves: ['darkpulse']},
+		], [
+			{species: 'absol', ability: 'pressure', moves: ['sleeptalk']},
+		]]);
+		battle.makeChoices('move darkpulse zmove', 'auto');
+		assert.equal(battle.p1.active[0].getMoveData(Dex.getMove('darkpulse')).pp, 22);
+	});
+
+	it.skip(`should deduct additional PP from submoves that target Pressure`, function () {
+		battle = common.createBattle([[
+			{species: 'wynaut', moves: ['assist']},
+			{species: 'yveltal', moves: ['darkpulse']},
+		], [
+			{species: 'absol', ability: 'pressure', moves: ['sleeptalk']},
+		]]);
+		battle.makeChoices('move assist', 'auto');
+		assert.equal(battle.p1.active[0].getMoveData(Dex.getMove('assist')).pp, 30);
+	});
+
+	it.skip(`Sticky Web should not deduct PP from Pressure (only entry hazard to do so)`, function () {
```

Comment:
Right. Imprison, Snatch, Spikes, Toxic Spikes, and Stealth Rock are all field-target moves explicitly assigned a "Pressure also works on these!", but Sticky Web didn't make it in the list :(

Reply:
(The sim actually considers Spikes, Toxic Spikes and Stealth Rock to be foe side-target moves rather than field-target moves, which is why the sim automatically calculates Pressure on those, but yes, it too has to special-case Snatch as well as Imprison.)

Reply:
Bah, I used the wrong language with "field-target"; they of course only target one side. But yeah, it's a hardcode either way probably.